app/models/membership_application.rb

```diff
@@ -136,13 +136,14 @@ def is_member?
   def accept_membership
     begin
 
-      membership_number = self.membership_number.blank? ? self.class.get_next_membership_number : self.membership_number
+      user.issue_membership_number
+      user.update(membership_number: membership_number)
```

Comment:
This might be clearer as:
```
user.save
```
Here, since we've delegated `membership_number`, this is the same as:
```
user.update(membership_number: user.membership_number)
```

Reply:
I realize I forgot to add a unique index to the `membership_number` field on the `user` table (needed for concurrent access, against which Rails cannot guard). I will add that together with the change @patmbolger suggested above.